### DISCUSSION TITLE
Setup the COBOL-DB2 compile and bind and CICS newcopy in DBB user build by IDz

### DISCUSSION THREAD

**Discussion by ratnambharti** [IBM/dbb-zappbuild#488] [Q&A]
I am customizing the zappbuild framework for my machine .I am trying to execute user build by IDz to build single cobol,COBOL-DB2, COBOL-DB2-CICS ,COBOL-IMS programs for each type.

My questions are-

1-How can I customized zAppbuild such as it will compile and bind the cobol-db2 program by IDz user build itself . and if it is cobol-cics-db2 then it should perform each step like compile,bind ,linkedit and also newcopy in cics region together with the user build operation only, What should be the changes required in zappbuild framework.I am not using any CI and CD tool and only  want to create deployable package by user build or by IDz and DBB . 

2-Can we build,packaged and deploy the multiple components without and CI tool like(Zenkins,Gitlab runner etc) and CD tool (UCD) . I want to use only DBB and IDz for packaging and deployment of multiple components 

Thanks a lot

**Comment by dennis-behm** [comment]
@ratnambharti 

1- Let me start with some core principles - building an application is independent of the runtime environment. zAppBuild and DBB are capable of dynamically assessing each file by its scanners to dynamically determine if a file contains CICS, DB2 statements. Based on that, the build framework is dynamically adding the correct allocations to SYLSIB or STEPLIB:

https://github.com/IBM/dbb-zappbuild/blob/9f2561711cb4adae85376b8513d69b4220f4a9f0/languages/Cobol.groovy#L231-L236

You can read more about it here - https://www.ibm.com/docs/en/dbb/2.0?topic=apis-dependency-impact-resolution#1-scanning-source-file-dependencies

Regarding db2 bin - there is a compromise to allow user builds to perform the Db2 bind step. Please look up the description of `bind_performBindPackage` property.

Please note that, build is building into **build** datasets, which are not meant to be added to the runtime systems. A packaging step like outlined in https://github.com/IBM/dbb/tree/main/Pipeline will help you to create a TAR file that can optionally be published to an Artefact repository. This package can also be deployed - via the deployment manager. Or implement your own script to take a shortcut to copy files into the runtime environment.

2- zAppBuild/DBB is a pure CLI driven tool and does not require a CI tool to run it. You can just use shell scripts to invoke it. Please have a look in the docs: https://github.com/IBM/dbb-zappbuild/blob/main/docs/BUILD.md for the various invocation samples.
